src/Microsoft.Experimental.Collections/Microsoft/Collections/Extensions/OrderedDictionary.cs

```diff
@@ -0,0 +1,981 @@
+        /// <exception cref="ArgumentNullException"><paramref name="key" /> is null.</exception>
+        public TValue GetOrAdd(TKey key, TValue value) => GetOrAdd(key, () => value);
+
+        /// <summary>
+        /// Adds a key/value pair to the <see cref="OrderedDictionary{TKey, TValue}" /> by using the specified function, if the key does not already exist as an O(1) operation.
+        /// </summary>
+        /// <param name="key">The key of the element to add.</param>
+        /// <param name="valueFactory">The function used to generate a value for the key.</param>
+        /// <returns>The value for the key. This will be either the existing value for the key if the key is already in the dictionary, or the new value for the key as returned by valueFactory if the key was not in the dictionary.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="key" /> is null.-or-<paramref name="valueFactory"/> is null.</exception>
+        public TValue GetOrAdd(TKey key, Func<TValue> valueFactory)
+        {
+            if (valueFactory == null)
+            {
+                throw new ArgumentNullException(nameof(valueFactory));
+            }
+
+            int index = IndexOf(key, out uint hashCode);
+            TValue value;
+            if (index < 0)
+            {
+                value = valueFactory();
+                AddInternal(null, key, value, hashCode);
+            }
+            else
+            {
+                value = _entries[index].Value;
+            }
+            return value;
+        }
+
+        /// <summary>
+        /// Gets the value associated with the specified key as an O(1) operation.
+        /// </summary>
+        /// <param name="key">The key of the value to get.</param>
+        /// <returns>The value associated with the specified key. If the specified key is not found, throws a <see cref="KeyNotFoundException" /></returns>
+        /// <exception cref="ArgumentNullException"><paramref name="key" /> is null.</exception>
+        /// <exception cref="KeyNotFoundException"><paramref name="key" /> does not exist in the collection.</exception>
+        public TValue GetValue(TKey key)
+        {
+            int index = IndexOf(key);
+            if (index < 0)
+            {
+                throw new KeyNotFoundException(string.Format(Strings.Arg_KeyNotFoundWithKey, key.ToString()));
+            }
+            return _entries[index].Value;
+        }
+
+        /// <summary>
+        /// Returns the zero-based index of the element with the specified key within the <see cref="OrderedDictionary{TKey, TValue}" /> as an O(1) operation.
+        /// </summary>
+        /// <param name="key">The key of the element to locate.</param>
+        /// <returns>The zero-based index of the element with the specified key within the <see cref="OrderedDictionary{TKey, TValue}" />, if found; otherwise, –1.</returns>
```

Comment:
`-1` on this line is actually `U+2013	EN DASH` followed by `1`. Can you please use regular ASCII hyphen or else the C# `\u2013`? We use `\u` for anything over 0x7F.

Reply:
Good catch. That was an issue in the source I copied from the XML Documentation on Dictionary.